forum/qa-plugin/user-activity-log/user-activity-log-search.php

```diff
@@ -0,0 +1,297 @@
+            }else{
+                $result = qa_db_query_sub($finalQuery, $this->searchArr['request'], $this->searchArr['date']."%", $this->searchArr['resultsCount']);
+            }
+        }else{
+            if(empty($this->searchArr['request'])){
+                $result = qa_db_query_sub($finalQuery, $this->searchArr['resultsCount']);
+            }else{
+                $result = qa_db_query_sub($finalQuery, $this->searchArr['request'], $this->searchArr['resultsCount']);
+            }
+        }
+       
+        if(mysqli_num_rows($result) <= $this->searchArr['resultsCount']){
+            $this->dbResult = qa_db_read_all_assoc($result);
+            return true;
+        }else{
+            $this->dbResult = $result;
+            return false;
+        }
+
+    }
+
+    private function generateResultsTable()
+    {
+        $unfiltered = $this->dbResult;
+        if(qa_get_logged_in_level() < QA_USER_LEVEL_ADMIN){
+            $results = array_filter($unfiltered, function($field){
+                $toExclude = [
+                    'q_vote_up',
+                    'q_vote_down', 
+                    'q_vote_nil',
+                    'a_vote_up',
+                    'a_vote_down', 
+                    'a_vote_nil', 
+                    'c_vote_up',
+                    'c_vote_down', 
+                    'c_vote_nil',
+                    'in_q_vote_up' ,
+                    'in_a_vote_up' ,
+                    'in_c_vote_up' ,
+                    'q_flag',
+                    'a_flag',
+                    'c_flag',
+                    'q_unflag',
+                    'a_unflag',
+                    'c_unflag',
+                    'q_clearflags',
+                    'a_clearflags',
+                    'c_clearflags',
+                ];
+                $isToExclude = false;
+                if(in_array($field['event'], $toExclude)){
+                    $isToExclude = true;
+                }
+
+                return $isToExclude ? "" : $field;
+            });
+        }else{
+            $results = $unfiltered;
+        }
+
+        $tableHeader = 
+        "<table><tr><th>".qa_lang_html('user-activity-log/datetime').' </th>'.
+            '<th>'.qa_lang_html('user-activity-log/handle').'</th>'.
+            '<th>'.qa_lang_html('user-activity-log/event').'</th>';
+
+        $tableContent = '';
+
+        if(qa_get_logged_in_level() > QA_USER_LEVEL_EDITOR){
+            $tableHeader = $tableHeader.'<th> '.qa_lang_html('user-activity-log/ipaddress').'</th>'.
+                "</tr>";
+
+            foreach($results as $row){
+                $tableContent = $tableContent.'<tr>'.
+                    '<td>'.$row['datetime'].'</td>'.
+                    '<td>'.$row['handle'].'</td>'.
+                    '<td>'.$this->findUsersPostsLinks($row['event'], $row['params']).'</td>'.
+                    '<td> &nbsp;'.$row['ipaddress'].'</td>'.
```

Comment:
Czemu służy tutaj `&nbsp;`? Jeśli trzeba nadać jakiś stylistyczny odstęp, to do tego służy CSS.

Reply:
Ten `&nbsp;` był potrzebny w jednej z poprzednich wersji. Bez niego, kolumny w tabeli się na siebie nakładały. Teraz jednak wygląda na to że i bez tego tabela jest dobrze ułożona